great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -190,7 +191,7 @@ def _get_data_reference_list_from_cache_by_data_asset_name(
         path_list: List[str] = [
             map_batch_definition_to_data_reference_string_using_regex(
                 batch_definition=batch_definition,
-                regex_pattern=pattern,
+                regex_pattern=re.compile(pattern),
```

Comment:
This is done so that the same utility method can support both, V3 as well as ZEP (where paths are handled more formally).